spec/lib/zipkin_kafka_tracer_spec.rb

```diff
@@ -26,24 +25,56 @@
     describe '#initialize' do
       context 'with default settings' do
 
+        before do
+          expect(Hermann::Discovery::Zookeeper).to receive(:new) { zk }
+          expect(zk).to receive(:get_brokers)
+          expect(Hermann::Producer).to receive(:new) { producer }
+        end
+
         it 'has default topic' do
           expect(tracer.instance_variable_get(:@topic)).to eq Trace::ZipkinKafkaTracer::DEFAULT_KAFKA_TOPIC
         end
 
-        it 'connects to zookeeper to create the producer' do
+        it 'connects to zookeeper to create a Hermann producer' do
           expect(tracer.instance_variable_get(:@producer)).to eq producer
         end
       end
 
-      context 'with options' do
+      context 'with options including a :topic' do
         let(:topic)  { 'topic' }
 
-        let(:tracer) { described_class.new({ topic: topic }) }
+        let(:tracer) { described_class.new({ :zookeepers => zookeepers, topic: topic }) }
 
         it 'has an optional topic' do
           expect(tracer.instance_variable_get(:@topic)).to eq topic
         end
       end
+
+      context 'with options including a :producer' do
+        subject { described_class.new(:producer => mock_producer) }
+
+        let(:mock_producer) { double('Kafka producer', :push => true) }
+
+        it 'should use that producer' do
```

Comment:
Usually we write in a more active way. Instead of `should use that producer` we do `uses that producer`. Same for other examples.